File: unity/Assets/Scripts/Multiplayer/Players/PlayerRegisterResponse.cs
using Michitai.Multiplayer.Errors;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Michitai.Multiplayer.Players
{
    /// <summary>
    /// Response returned when a new player is successfully registered.
    /// Contains the player's ID, private key for authentication, and game information.
    /// </summary>
    [System.Serializable]
    public class PlayerRegisterResponse : ApiResponse<EPlayerRegisterError>
    {
        /// <summary>
        /// The unique ID assigned to the newly registered player.
        /// </summary>
        public int player_id;

        /// <summary>
        /// The private key token used for player authentication.
        /// Keep this secure as it's required for all player-specific operations.
        /// </summary>
        public string private_key;

        /// <summary>
        /// The player's display name.
        /// </summary>
        public string player_name;

        /// <summary>
        /// The ID of the game the player was registered to.
        /// </summary>
        public int game_id;
    }
}

File: java/src/main/java/com/michitai/multiplayer/players/PlayerRegisterResponse.java
package com.michitai.multiplayer.players;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.michitai.multiplayer.ApiResponse;

/**
 * Response returned when a new player is successfully registered.
 * Contains the player's ID, private key for authentication, and game information.
 */
public class PlayerRegisterResponse extends ApiResponse {
    @JsonProperty("player_id")
    private int playerId;

    @JsonProperty("private_key")
    private String privateKey;

    @JsonProperty("player_name")
    private String playerName;

    @JsonProperty("game_id")
    private int gameId;

    public int getPlayerId() {
        return playerId;
    }

    public void setPlayerId(int playerId) {
        this.playerId = playerId;
    }

    public String getPrivateKey() {
        return privateKey;
    }

    public void setPrivateKey(String privateKey) {
        this.privateKey = privateKey;
    }

    public String getPlayerName() {
        return playerName;
    }

    public void setPlayerName(String playerName) {
        this.playerName = playerName;
    }

    public int getGameId() {
        return gameId;
    }

    public void setGameId(int gameId) {
        this.gameId = gameId;
    }
}
